### ISSUE TITLE
AbiIO.py issue with reading file from .fsa file: seq referenced before assignment

### ISSUE THREAD

**Issue by graph1994** [biopython/biopython#1030]
I'm using SeqIO.read to read a fsa file from an Applied Biosystems 3730xl machine.
Fails to find the key of PBAS2 then doesn't set seq thus causing it to fail when used on line 411. I've found if I hard code the seq, smaple_id and qual array it is able to build with abif_raw intact and usable. 

3730xl does not have PBAS2 listed: [ABI Format docs](http://www6.appliedbiosystems.com/support/software_community/ABIF_File_Format.pdf)
I would like to extend this file to allow for 3730xl 

This is how I'm using it: `record = SeqIO.read(file, 'abi')`

Stack Trace:
```
Traceback (most recent call last):
  File "/Users/kgraff/Projects/fsa_interperter/fsa_reader.py", line 48, in <module>
    result = openFileCreatePlot(fileNameDir)
  File "/Users/kgraff/Projects/fsa_interperter/fsa_reader.py", line 20, in openFileCreatePlot
    record = SeqIO.read(file, 'abi')
  File "/Users/kgraff/anaconda/lib/python2.7/site-packages/Bio/SeqIO/__init__.py", line 664, in read
    first = next(iterator)
  File "/Users/kgraff/anaconda/lib/python2.7/site-packages/Bio/SeqIO/__init__.py", line 600, in parse
    for r in i:
  File "/Users/kgraff/anaconda/lib/python2.7/site-packages/Bio/SeqIO/AbiIO.py", line 411, in AbiIterator
    record = SeqRecord(Seq(seq, alphabet),
UnboundLocalError: local variable 'seq' referenced before assignment
```

**Comment by peterjc** [comment]
@bow could you take a look at this please?

@graph1994 could you share a test file for debugging, ideally something we could include in future releases for our unit tests?

Thanks!

**Comment by graph1994** [comment]
If you don't mind I'd like to try and help solve the problem, I'll try to find a test file that can be used and include that with my PR as well

**Comment by peterjc** [comment]
Sure - Bow wrote the original code so would be the ideal person to review a pull request from you.

**Comment by bow** [comment]
@graph1994, thanks for reporting the bug and feel free to explore how to solve the problem. The current parser does lack tests for .fsa files, so adding them would be much appreciated.

**Comment by bow** [comment]
Closing this issue since this has been addressed in #1037.
